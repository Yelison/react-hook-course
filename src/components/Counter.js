import React from "react";
import PropTypes from "prop-types";

const Counter = ({ value }) => {
  const handleClick = () => {
    console.log("+1");
  };

  return (
    <>
      <h1>Counter</h1>
      <div>{ value }</div>
      <button onClick={ handleClick }>Click me!</button>
    </>
  );
};

Counter.propTypes = {
  value: PropTypes.number,
};

Counter.defaultProps = {
  value: 0,
};

export default Counter;
