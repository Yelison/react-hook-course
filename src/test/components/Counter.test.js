import React from 'react';
import { shallow } from 'enzyme';
import Counter from '../../components/Counter';

describe('Probando el componente Counter', () => {
    
    test('Probando  que el <Counter /> se muestre correctamente', () => {
        
      const wrapper = shallow(<Counter />);
      
      expect( wrapper ).toMatchSnapshot();

    })
    

})
