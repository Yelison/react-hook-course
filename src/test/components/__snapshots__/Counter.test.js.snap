// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probando el componente Counter Probando  que el <Counter /> se muestre correctamente 1`] = `
<Fragment>
  <h1>
    Counter
  </h1>
  <div>
    0
  </div>
  <button
    onClick={[Function]}
  >
    Click me!
  </button>
</Fragment>
`;
